sklego/model_selection.py

```diff
@@ -485,8 +472,10 @@ def _iter_test_indices(self, X=None, y=None, groups=None):
         self._first_split_index, self._last_split_index = self._calc_first_and_last_split_index(groups=groups)
         self._best_splits = self._get_split_indices()
         groups = self._regroup(groups)
+        group_indices = tuple(np.where(groups == i)[0] for i in range(self.n_splits + 1))
+
         for i in range(self.n_splits):
-            yield np.where(groups == i)[0], np.where(groups == i + 1)[0]
+            yield group_indices[i], group_indices[i + 1]
 
     def _calc_first_and_last_split_index(self, X=None, y=None, groups=None):
```

Comment:
In a way, the changes in `_calc_first_and_last_split_index` are the only hard core difference